utility/utils.py

```diff
@@ -715,47 +714,136 @@ def destroy_cluster(cluster_path):
     """
-    Download the openshift installer
+    Download the OpenShift installer binary, if not already present.
+    Update env. PATH and get path of the openshift installer binary.
 
     Args:
         version (str): version of the installer to download
 
     Returns:
-        str: name of the installer binary after being unpacked
+        str: path of the installer binary
 
     """
     installer_filename = "openshift-install"
-    tarball = f"{installer_filename}.tar.gz"
-    if os.path.isfile(installer_filename):
-        log.info("Installer exists, skipping download")
+    installer_binary_path = os.path.join(defaults.BIN_DIR, installer_filename)
+    if os.path.isfile(installer_binary_path):
+        log.debug("Installer exists ({installer_binary_path}), skipping download.")
+        # TODO: check installer version
     else:
         log.info("Downloading openshift installer")
-        if platform.system() == "Darwin":
-            os_type = "mac"
-        elif platform.system() == "Linux":
-            os_type = "linux"
-        else:
-            raise UnsupportedOSType
-        url = (
-            f"https://mirror.openshift.com/pub/openshift-v4/clients/ocp/"
-            f"{version}/openshift-install-{os_type}-{version}.tar.gz"
-        )
+        prepare_bin_dir()
+        # record current working directory and switch to BIN_DIR
+        previous_dir = os.getcwd()
+        os.chdir(defaults.BIN_DIR)
+        tarball = f"{installer_filename}.tar.gz"
+        url = get_openshift_mirror_url(installer_filename, version)
         download_file(url, tarball)
         run_cmd(f"tar xzvf {tarball}")
+        os.remove(tarball)
+        # return to the previous working directory
+        os.chdir(previous_dir)
+
+    add_path_to_env_path(defaults.BIN_DIR)
+
+    return installer_binary_path
+
+
+def get_openshift_client(version=defaults.CLIENT_VERSION):
+    """
+    Download the OpenShift client binary, if not already present.
+    Update env. PATH and get path of the oc binary.
+
+    Args:
+        version (str): version of the client to download
+
+    Returns:
+        str: path of the client binary
+
+    """
+    client_binary_path = os.path.join(defaults.BIN_DIR, 'oc')
+    if os.path.isfile(client_binary_path):
+        log.debug("Client exists ({client_binary_path}), skipping download.")
+        # TODO: check client version
+    else:
+        log.info("Downloading openshift client")
+        prepare_bin_dir()
+        # record current working directory and switch to BIN_DIR
+        previous_dir = os.getcwd()
+        os.chdir(defaults.BIN_DIR)
+        url = get_openshift_mirror_url('openshift-client', version)
+        tarball = "openshift-client.tar.gz"
+        download_file(url, tarball)
+        run_cmd(f"tar xzvf {tarball}")
+        os.remove(tarball)
+        # return to the previous working directory
+        os.chdir(previous_dir)
+
+    add_path_to_env_path(defaults.BIN_DIR)
```

Comment:
This should not be hidden there, but called at the same level where `get_openshift_client()` is called.

Also: we need to make sure `PATH` is configured properly for tests executed on already provisioned cluster (when ` tests/test_ocs_basic_install.py` is not executed). That said, since we are moving to pytest, we could just note this to be a known issue and solve it with pytest fixture properly after this pull request is merged.

Reply:
My intention with this approach was, that anybody who needs `oc` command will call the `get_openshift_client()` function, which will ensure that the client is available and the use either the `oc` command rely on the PATH configured by the function or will use the full path to `oc` command returned by the function.